### PR TITLE
Expose new Stripe subscription `cancellation_reason` metadata

### PR DESCRIPTION
**Pull request by sean-rose**
On 2024-03-06 SubPlat started adding a `cancellation_reason` metadata field on production Stripe subscriptions when it preemptively cancels them ([FXA-9067](https://mozilla-hub.atlassian.net/browse/FXA-9067)).

Once this PR is merged I will backfill the associated incremental ETLs back to 2024-03-06.

---
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

[FXA-9067]: https://mozilla-hub.atlassian.net/browse/FXA-9067?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3000)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_subscriptions_revised_changelog_v1/query.sql [hunk 1]
```diff
@@ -97,6 +97,7 @@ WITH original_changelog AS (
         subscription.* REPLACE (
           STRUCT(
             JSON_VALUE(subscription.metadata.appliedPromotionCode) AS appliedPromotionCode,
+            JSON_VALUE(subscription.metadata.cancellation_reason) AS cancellation_reason,
```
Comment (BenWu) [inline sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_subscriptions_revised_changelog_v1/query.sql:100]: Does cancellation_reason need to be added to the metadata struct in the synthesize_subscription udf?  I don't see why it wouldn't be but the query is tough to follow

Comment (sean-rose) [reply sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_subscriptions_revised_changelog_v1/query.sql:100]: Yes, it should be added there as well. Good catch!

**Review by BenWu — APPROVED** [review]
What's the difference between the v1 and v2 of the subscriptions tables and why isn't this field needed in the v1 tables?  Are v1 tables going to be deprecated?

The required check is still failing but it's unrelated so lgtm

**Comment by sean-rose** [comment]
> What's the difference between the v1 and v2 of the subscriptions tables and why isn't this field needed in the v1 tables? Are v1 tables going to be deprecated?

v2 is the newest version which I created to support product agnostic subscriptions ETLs (i.e. "SubPlat consolidated reporting").

v1 was originally created to support the VPN subscriptions ETLs, and also supports the Relay and Hubs subscriptions ETLs (which were basically copies of the VPN ETLs).  It will indeed be deprecated when the dashboards for VPN and Relay are migrated to the new SubPlat consolidated reporting setup.

**Pull request merged**